python/lsst/source/injection/inject_coadd.py

```diff
@@ -82,3 +129,124 @@ def runQuantum(self, butler_quantum_context, input_refs, output_refs):
         input_keys = ["injection_catalogs", "input_exposure", "sky_map", "psf", "photo_calib", "wcs"]
         outputs = self.run(**{key: value for (key, value) in inputs.items() if key in input_keys})
         butler_quantum_context.put(outputs, output_refs)
+
+    def get_variance_scale(self, exposure):
+        x = exposure.image.array.flatten()
+        y = exposure.variance.array.flatten()
+
+        # Identify bad pixels
+        bad_pixels = ~np.isfinite(x) | ~np.isfinite(y)
+        # Replace bad pixel values with the image median
+        if np.sum(bad_pixels) > 0:
+            median_image_value = np.median(x)
+            median_variance_value = np.median(y)
+            x[bad_pixels] = median_image_value
+            y[bad_pixels] = median_variance_value
+
+        # Simple linear regression to establish MSE.
+        linear = LinearRegression()
+        linear.fit(x.reshape(-1, 1), y)
+        linear_mse = mean_squared_error(y, linear.predict(x.reshape(-1, 1)))
+
+        # First RANSAC fit
+        fit_results = []
+        for seed in range(
+            self.config.variance_fit_seed_1, self.config.variance_fit_seed_1 + self.config.n_fits_1
+        ):
+            ransac = RANSACRegressor(
+                loss="squared_error",
+                residual_threshold=self.config.threshold_scale_1 * linear_mse,
+                max_trials=1000,
```

Comment:
Should this hard-coded value be accessible at the function keyword level, or even the Task config level?

Reply:
It makes sense to put it at the function keyword level at least, and I'm fine with putting it in the Task config if people are happy with the proliferation of config options. Ideally this value should be high enough so that it's never the thing that causes a pathology in practice, and it seems like scikit-learn's default value is a little low, which is why I specify a different value here.